velox/exec/tests/ExchangeFuzzer.cpp

```diff
@@ -0,0 +1,452 @@
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/parse/TypeResolver.h"
+#include "velox/serializers/PrestoSerializer.h"
+#include "velox/vector/VectorSaver.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include <fstream>
+
+DEFINE_int32(max_tasks_per_stage, 16, "Max number of sources/destinations");
+DEFINE_int32(drivers_per_task, 4, "Number of threads in each task in shuffle");
+DEFINE_int64(shuffle_bytes, 4UL << 30, "Shuffle data volume in each step");
+DEFINE_int32(max_buffer_mb, 20, "Max buffer size for output/exchange per task");
+DEFINE_uint64(seed, 0, "Seed, 0 means random");
+
+DEFINE_int32(steps, 10, "Number of plans to generate and test.");
+
+DEFINE_int32(duration_sec, 0, "Run duration in seconds");
+
+DEFINE_bool(inject_failure, false, "Inject a failure for testing repro");
+
+DEFINE_string(
+    repro_path,
+    ".",
+    "Path for writing repro files in case of failure");
+
+DEFINE_string(
+    replay,
+    "",
+    "File to replay. Files are produced on failure in --repro_path");
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::test;
+
+class ExchangeFuzzer : public VectorTestBase {
+ public:
+  ExchangeFuzzer() : fuzzer_(options_, pool_.get()) {}
+
+  bool runOne(
+      std::vector<RowVectorPtr>& vectors,
+      int32_t numSourceTasks,
+      int32_t numDestinationTasks,
+      int32_t numDriversPerTask,
+      int32_t outputBufferBytes,
+      int32_t exchangeBufferBytes,
+      int32_t batchBytes) {
+    assert(!vectors.empty());
+    auto iteration = ++iteration_;
+    LOG(INFO) << "Iteration " << iteration << " shuffl " << numSourceTasks
+              << "x" << numDestinationTasks << " drivers=" << numDriversPerTask
+              << " Type " << vectors.front()->type()->toString()
+              << " output buffer=" << outputBufferBytes
+              << " exchange buffer=" << exchangeBufferBytes
+              << " target batch=" << batchBytes;
+    configSettings_[core::QueryConfig::kMaxPartitionedOutputBufferSize] =
+        fmt::format("{}", outputBufferBytes);
+    configSettings_[core::QueryConfig::kMaxExchangeBufferSize] =
+        fmt::format("{}", exchangeBufferBytes);
+    configSettings_[core::QueryConfig::kPreferredOutputBatchBytes] =
+        fmt::format("{}", batchBytes);
+
+    auto& rowType = vectors.front()->type()->as<TypeKind::ROW>();
+    std::vector<std::string> aggregates;
+    std::vector<std::vector<TypePtr>> rawInputTypes;
+    for (auto i = 1; i < rowType.size(); ++i) {
+      rawInputTypes.push_back({rowType.childAt(i)});
+    }
+    auto expected = expectedChecksums(
+        vectors, rawInputTypes, numSourceTasks * numDriversPerTask);
+
+    std::vector<std::shared_ptr<Task>> tasks;
```

Comment:
this method is quite long. Can we move the task creation and set up to a separate helper function?